### DISCUSSION TITLE
Accessing Context Bridge event in Vue component with typescript

### DISCUSSION THREAD

**Discussion by tgarif** [quasarframework/quasar#10160] [CLI - Electron mode] [answered]
I have defined some events in my preload-script.ts

![preload](https://user-images.githubusercontent.com/60260466/126590197-2e1b4132-e40c-4347-a130-72913127e4cf.PNG)

When I try to access it from my Vue component, it gaves me warning

![main](https://user-images.githubusercontent.com/60260466/126590404-b14a3e90-2856-4d3b-9dc2-880af054aa2c.PNG)

Unsafe member access .toggleMaximize on an `any` value.
Unsafe call of an `any` typed value.
Property 'myWindowAPI' does not exist on type 'Window & typeof globalThis'.

Any idea how to solve this?

**Comment by MicheleQuasar** [comment]
Had the same problem, I think I managed to solve the issue with the compiler error but **not** with the syntax checker or intellisense in VSCode.

According to the electron documentation (https://www.electronjs.org/docs/v14-x-y/tutorial/context-isolation#usage-with-typescript), to make context isolation work with TypeScript you need to add a types declaration file to augment the defult `Window` interface with the API injected from `contextBridge`. I found that this for some reason does not work for me, what I needed to do was follow this (https://stackoverflow.com/questions/57132428/augmentations-for-the-global-scope-can-only-be-directly-nested-in-external-modul) stackoverflow question and split the declaration file in two parts.

Example:

say you want to a expose a simple API via contextBridge
```
//src-electron/electron-preload.ts
import { contextBridge } from 'electron'

contextBridge.exposeInMainWorld('myAPI', {
    hello: 'World!'
})
```
then write **two** declaration files:
```
//src-electron/electron-preload.d.ts

declare interface ImyAPI {
    hello: string
}
```
```
//src/global.d.ts

export {};

declare global{
    interface Window {
        myAPI: ImyAPI
    }
}
```

For some reason, it looks like the empty `export` in the second declaration file is necessary.

Note:
- Looks like this is not an optimal solution, just noticed that closing and reopening the project I have to compile twice (i.e. complie, stop the running window and comiple again) for the compiler to notice the declaration file. I have no idea why.
- I indicated paths for the files but i'm not sure they are in the best/correct location for everything to work smoothly.

**Comment by hawkeye64** [comment, marked as answer]
You can also go into `.eslintrc` and add your namespace to the globals section.
